### PR TITLE
org, team and project entities implementation

### PR DESCRIPTION
**Pull request by kmrdhruv**
TODO::
UTs will be added post initial review.

### CODE REVIEW

File: server/src/main/java/com/flipkart/varadhi/db/VaradhiMetaStore.java [hunk 1]
```diff
@@ -1,66 +1,251 @@
...
     @Override
-    public boolean checkVaradhiTopicExists(String varadhiTopicName) {
-        String resourcePath = ZKPathUtils.getVaradhiTopicPath(varadhiTopicName);
-        return this.zkMetaStore.exists(resourcePath);
+    public List<Org> getOrgs() {
+        ZNode znode = ZNode.OfEntityType(ORG);
+        return this.zkMetaStore.listChildren(znode)
+                .stream()
+                .map(name -> {
+                    //TODO:: Discuss for Removing this handling. Discuss this exception handling may not be required if
```
Comment (gauravAshok) [inline server/src/main/java/com/flipkart/varadhi/db/VaradhiMetaStore.java:68]: yes, fair assumption. we can fail the whole operation even if one element fails.

File: server/src/main/java/com/flipkart/varadhi/web/v1/OrgHandlers.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import static com.flipkart.varadhi.auth.ResourceAction.ORG_DELETE;
+import static com.flipkart.varadhi.auth.ResourceAction.ORG_GET;
+import static com.flipkart.varadhi.web.routes.RouteBehaviour.authenticated;
+import static com.flipkart.varadhi.web.routes.RouteBehaviour.hasBody;
+
+
+@Slf4j
+@ExtensionMethod({Extensions.RequestBodyExtension.class, Extensions.RoutingContextExtension.class})
+public class OrgHandlers implements RouteProvider {
+    //TODO:: Discuss case preserving or lower case requirement and close this again.
```
[the remark this answers is not in the document]
Comment (kmrdhruv) [reply server/src/main/java/com/flipkart/varadhi/web/v1/OrgHandlers.java:29]: let's discuss.  Case insensitive/preserving or lower case ?
I guess implementation was implementing single case would be much easier and less prone than case insensitive or preserving semantics.

File: entities/src/main/java/com/flipkart/varadhi/entities/Project.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+@Getter
+@EqualsAndHashCode(callSuper = true)
+public class Project extends VaradhiResource {
+    String orgName;
+
+    @Setter
+    String teamName;
+
+    @Setter
+    String description;
```
[the remark this answers is not in the document]
Comment (kmrdhruv) [reply entities/src/main/java/com/flipkart/varadhi/entities/Project.java:16]: these are mutable attributes or u mean something else ?

File: server/src/main/java/com/flipkart/varadhi/services/ProjectService.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    public Project createProject(Project project) {
+        boolean orgExists = this.metaStore.checkOrgExists(project.getOrgName());
+        if (!orgExists) {
+            throw new ResourceNotFoundException(String.format(
+                    "Org(%s) not found. For Project creation, associated Org and Team should exist.",
+                    project.getOrgName()
+            ));
+        }
+        boolean teamExists = this.metaStore.checkTeamExists(project.getTeamName(), project.getOrgName());
```
[the remark this answers is not in the document]
Comment (kmrdhruv) [reply server/src/main/java/com/flipkart/varadhi/services/ProjectService.java:29]: correct. It was not there.. added it to give higher order error first.

**Pull request merged**